discovery-provider/src/queries/query_helpers.py

```diff
@@ -371,40 +282,28 @@ def get_track_play_count_dict(session, track_ids):
 #   if current_user_id available, populates followee_reposts, has_current_user_reposted, has_current_user_saved
 def populate_track_metadata(session, track_ids, tracks, current_user_id):
     # build dict of track id --> repost count
-    repost_counts = (
+    counts = (
         session.query(
-            Repost.repost_item_id,
-            func.count(Repost.repost_item_id)
+            AggregateTrack.track_id,
+            AggregateTrack.repost_count,
+            AggregateTrack.save_count
         )
         .filter(
-            Repost.is_current == True,
-            Repost.is_delete == False,
-            Repost.repost_item_id.in_(track_ids),
-            Repost.repost_type == RepostType.track
+            AggregateTrack.track_id.in_(track_ids),
         )
-        .group_by(Repost.repost_item_id)
         .all()
     )
-    repost_count_dict = {track_id: repost_count for (
-        track_id, repost_count) in repost_counts}
 
-    # build dict of track id --> save count
-    save_counts = (
-        session.query(
-            Save.save_item_id,
-            func.count(Save.save_item_id)
-        )
-        .filter(
-            Save.is_current == True,
-            Save.is_delete == False,
-            Save.save_type == SaveType.track,
-            Save.save_item_id.in_(track_ids)
-        )
-        .group_by(Save.save_item_id)
-        .all()
-    )
-    save_count_dict = {track_id: save_count for (
-        track_id, save_count) in save_counts}
+    count_dict = {
+        track_id: {
+            response_name_constants.repost_count: repost_count,
+            response_name_constants.save_count: save_count
+        } for (
+            track_id,
+            repost_count,
+            save_count
+        ) in counts
+    }
 
     play_count_dict = get_track_play_count_dict(session, track_ids)
```

Comment:
i think getting plays into this matview would be worthwhile, but can leave off for now. the track matview could just use the aggregate_plays matview directly